dll/cpl/usrmgr/users.c

```diff
@@ -650,6 +651,45 @@ UpdateUserProperties(HWND hwndDlg)
     NetApiBufferFree(pUserInfo);
 }
 
+void OnToggleRequireLogon(HWND hwndDlg)
+{
+    BOOL bIsChecked;
+    WCHAR szAutoAdminLogonValue[2]; 
+    HKEY hKey;
+    LONG lResult;
+
+    bIsChecked = IsDlgButtonChecked(hwndDlg, IDC_USERS_STARTUP_REQUIRE);
+
+    wcscpy(szAutoAdminLogonValue, (bIsChecked == BST_CHECKED) ? L"0" : L"1");
+
+    lResult = RegCreateKeyExW(HKEY_LOCAL_MACHINE,
+                              L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
+                              0,
+                              NULL,
+                              REG_OPTION_NON_VOLATILE,
+                              KEY_SET_VALUE,
+                              NULL,
+                              &hKey,
+                              NULL);
+    if (lResult != ERROR_SUCCESS)
+    {
+        DPRINTF("OnToggleRequireLogon: Failed to open or create Winlogon registry key. Error code: %ld\n", lResult);
+        return; 
+    }
+
+    lResult = RegSetValueExW(hKey,
+                             L"AutoAdminLogon",
+                             0,
+                             REG_SZ, 
+                             (const BYTE*)szAutoAdminLogonValue,
+                             sizeof(szAutoAdminLogonValue));
```

Comment:
```suggestion
                             2 * sizeof(WCHAR));
```

Reply:
I don't follow: `sizeof()` cannot be applied to a pointer (`pszAutoAdminLogonValue`)...